src/pool/pool_scalable.c

```diff
@@ -109,46 +114,60 @@ static const char *tbb_symbol[TBB_POOL_SYMBOLS_MAX] = {
+    *(void **)&tbb_callbacks.pool_aligned_malloc =
+        utils_get_symbol_addr(tbb_callbacks.lib_handle,
                               tbb_symbol[TBB_POOL_ALIGNED_MALLOC], lib_name);
-    *(void **)&tbb_callbacks->pool_free = utils_get_symbol_addr(
-        tbb_callbacks->lib_handle, tbb_symbol[TBB_POOL_FREE], lib_name);
-    *(void **)&tbb_callbacks->pool_create_v1 = utils_get_symbol_addr(
-        tbb_callbacks->lib_handle, tbb_symbol[TBB_POOL_CREATE_V1], lib_name);
-    *(void **)&tbb_callbacks->pool_destroy = utils_get_symbol_addr(
-        tbb_callbacks->lib_handle, tbb_symbol[TBB_POOL_DESTROY], lib_name);
-    *(void **)&tbb_callbacks->pool_identify = utils_get_symbol_addr(
-        tbb_callbacks->lib_handle, tbb_symbol[TBB_POOL_IDENTIFY], lib_name);
-    *(void **)&tbb_callbacks->pool_msize = utils_get_symbol_addr(
-        tbb_callbacks->lib_handle, tbb_symbol[TBB_POOL_MSIZE], lib_name);
-
-    if (!tbb_callbacks->pool_malloc || !tbb_callbacks->pool_realloc ||
-        !tbb_callbacks->pool_aligned_malloc || !tbb_callbacks->pool_free ||
-        !tbb_callbacks->pool_create_v1 || !tbb_callbacks->pool_destroy ||
-        !tbb_callbacks->pool_identify) {
+    *(void **)&tbb_callbacks.pool_free = utils_get_symbol_addr(
+        tbb_callbacks.lib_handle, tbb_symbol[TBB_POOL_FREE], lib_name);
+    *(void **)&tbb_callbacks.pool_create_v1 = utils_get_symbol_addr(
+        tbb_callbacks.lib_handle, tbb_symbol[TBB_POOL_CREATE_V1], lib_name);
+    *(void **)&tbb_callbacks.pool_destroy = utils_get_symbol_addr(
+        tbb_callbacks.lib_handle, tbb_symbol[TBB_POOL_DESTROY], lib_name);
+    *(void **)&tbb_callbacks.pool_identify = utils_get_symbol_addr(
+        tbb_callbacks.lib_handle, tbb_symbol[TBB_POOL_IDENTIFY], lib_name);
+    *(void **)&tbb_callbacks.pool_msize = utils_get_symbol_addr(
+        tbb_callbacks.lib_handle, tbb_symbol[TBB_POOL_MSIZE], lib_name);
+
+    if (!tbb_callbacks.pool_malloc || !tbb_callbacks.pool_realloc ||
+        !tbb_callbacks.pool_aligned_malloc || !tbb_callbacks.pool_free ||
+        !tbb_callbacks.pool_create_v1 || !tbb_callbacks.pool_destroy ||
+        !tbb_callbacks.pool_identify) {
         LOG_FATAL("Could not find all TBB symbols in %s", lib_name);
-        utils_close_library(tbb_callbacks->lib_handle);
-        return -1;
+        if (utils_close_library(tbb_callbacks.lib_handle)) {
+            LOG_ERR("Could not close %s library", lib_name);
+        }
+        tbb_init_result = -1;
     }
+}
+
+static int init_tbb_callbacks(void) {
+    utils_init_once(&tbb_initialized, init_tbb_callbacks_once);
+    return tbb_init_result;
+}
 
-    return 0;
+void fini_tbb_global_state(void) {
+    if (tbb_callbacks.lib_handle) {
+        if (!utils_close_library(tbb_callbacks.lib_handle)) {
+            tbb_callbacks.lib_handle = NULL;
+            LOG_DEBUG("TBB library closed");
+        } else {
+            LOG_DEBUG("TBB library cannot be unloaded");
```

Comment:
nit: should it be a higher severity level, `LOG_ERROR`?

Reply:
👍